feat: Pandas and GeoPandas samples

jimfulton opened the pull request:
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-sqlalchemy/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #231 🦕

Commit:
Set up snippets

Commit:
first wrong stab at README

Commit:
Revert "first wrong stab at README"

This reverts commit 2e1fc9a42ec86ed1e8eada0d6d763da1d4fd12b6.

README generation is broke,
https://github.com/googleapis/synthtool/issues/867

so we'll punt for now.

Commit:
typo

Commit:
Add geopandas example

Commit:
Merge branch 'master' into pandas-samples

Commit:
Added the missing CRS argument.

Commit:
Merge branch 'pandas-samples' of github.com:jimfulton/python-bigquery-sqlalchemy into pandas-samples

Commit:
Added START and END, but I don't know what I'm doing. :)

jimfulton commented:
I added a "do not merge" tag, because I'm not sure we want people to do this.

However, I'd kind of like to know how to do samples, and would like some feedback. :)

gcf-owl-bot[bot] committed:
🦉 Updates from OwlBot

See https://github.com/googleapis/repo-automation-bots/blob/master/packages/owl-bot/README.md

tswast committed:
Merge branch 'master' into pandas-samples

tswast commented on samples/snippets/pandas_read_sql.py:
```diff
@@ -0,0 +1,53 @@
+
+
+def read_data_into_pandas_using_read_sql() -> None:
+    # [START sqlalchemy_bigquery_read_sql]
```
Needs to start with `bigquery`, how about `bigquery_pandas_sqlalchemy`?

jimfulton replied:
fixed

tswast reviewed (commented):
These look good! I think we just need to adjust the tags so that the snippet bot is happy.

tseaver commented on samples/snippets/geopandas_read_sql.py:
```diff
@@ -0,0 +1,78 @@
+        "EPSG:4326",
+    )
+
+    # Don't wrap pr elide:
```
I'm betting it is really `or elide:`.

jimfulton replied:
Yup, fixed

Commit:
typo

Commit:
fixed snippet tags

jimfulton commented on samples/snippets/geopandas_read_sql.py:
```diff
@@ -0,0 +1,78 @@
+
+
+def read_geographic_data_into_pandas_using_read_sql() -> None:
+    # [START sqlalchemy_bigquery_read_postgis]
```
fixed

tswast commented:
Kokoro failure looks like a flake by the backend. I learned that the backend can take as long as 4 minutes to fail on some API requests, so hopefully https://github.com/googleapis/python-bigquery/pull/859 helps with this.

```
google.api_core.exceptions.RetryError: Deadline of 120.0s exceeded while calling functools.partial(functools.partial(<bound method JSONConnection.api_request of <google.cloud.bigquery._http.Connection object at 0x7fe2a9e0e970>>, method='GET', path='/projects/precise-truck-742/queries/0aba5d5e-abdb-415a-9ffd-ad2aa450ad59', query_params={'maxResults': 0, 'location': 'US'}, timeout=None)), last exception: 500 GET https://bigquery.googleapis.com/bigquery/v2/projects/precise-truck-742/queries/0aba5d5e-abdb-415a-9ffd-ad2aa450ad59?maxResults=0&location=US&prettyPrint=false: An internal error occurred and the request could not be completed.
```

In the meantime, I'll manually trigger the Kokoro job to rebuild.

jimfulton commented:
> Kokoro failure looks like a flake by the backend. 

Yup, I figured.

...

> In the meantime, I'll manually trigger the Kokoro job to rebuild.

Cool, but remember the goal of this PR, at least for me, was to get a grip on samples. :)

It's been super helpful, but I don't intend to merge it.

IMO, these samples are counter productive, because, as you've noted, the BQ python libary's `to_dataset` (and soon, `to_geodataset`) is more performant and scalable.  Of course, if you disagree, I'm happy to defer.

My thought is when `to_geodataset` lands (https://github.com/googleapis/python-bigquery/pull/848 is ready for review, BTW :) ), I'd make a GeoPandas PR to add `from_bigquery` and `to_bigquery` using `to_geodataframe` and `load_from_dataframe`.

jimfulton commented:
I've merged this into the geography branch and deleted these two examples (and added a geography sample).

If you really want _these_ samples, let me know and I'll reopen this.
